stan/math/prim/prob/multi_normal_cholesky_lpdf.hpp

```diff
@@ -119,41 +120,120 @@ return_type_t<T_y, T_loc, T_covar> multi_normal_cholesky_lpdf(
     logp += NEG_LOG_SQRT_TWO_PI * size_y * size_vec;
   }
 
-  const matrix_partials_t inv_L_dbl
-      = mdivide_left_tri<Eigen::Lower>(value_of(L_ref));
+  if (include_summand<propto, T_y, T_loc, T_covar_elem>::value
+      && is_constant<T_covar_elem>::value) {
+    matrix_partials_t L_val = value_of(L_ref);
+
+    if (size_vec > 1) {
```

Comment:
Since `size_vec` is 

```cpp
const size_t size_vec = max_size_mvt(y, mu);
```

which just returns the max `size_mvt(...)` which is just 

```cpp
template <typename ScalarT, require_stan_scalar_t<ScalarT>* = nullptr>
size_t size_mvt(const ScalarT& /* unused */) {
  throw std::invalid_argument("size_mvt passed to an unrecognized type.");
}

template <typename MatrixT, require_matrix_t<MatrixT>* = nullptr>
size_t size_mvt(const MatrixT& /* unused */) {
  return 1U;
}

template <typename MatrixT, require_matrix_t<MatrixT>* = nullptr>
size_t size_mvt(const std::vector<MatrixT>& x) {
  return x.size();
}
```

The only time size_vec is going to be greater than 1 is when either `y` or `mu` is an `std::vector<Matrix>`. I think it would clean up this code a lot if it was broken up into two separate functions, one for when either `y` or `mu` is a vector of matrices and the other for when neither of them are. That's also nice because for the second function we'll know they are both just matrices and so also don't need to use `vector_seq_view`

Reply:
That's the situation. Yeah I can break these up into two functions.